src/modules/flow/accelerometer/adxl45.c

```diff
@@ -0,0 +1,401 @@
+         * time of testing, if too much time passes between two
+         * consecutive readings, the buffer will be reported full, but
+         * the last readings will contain trash values -- this will
+         * guard against that (one will have to read again to get
+         * newer values)*/
+#define MAX_EPSILON (10.0f)
+#define EPSILON_CHECK(_axis) \
+    if (i > 0 && isgreater(fabs((buffer[i][_axis] * ACCEL_SCALE_M_S) \
+        - mdata->reading[_axis]), MAX_EPSILON)) \
+        break
+
+        /* raw readings, with only the sensor-provided filtering */
+        for (int i = 0; i < num_samples_available; i++) {
+            EPSILON_CHECK(0);
+            EPSILON_CHECK(1);
+            EPSILON_CHECK(2);
+
+            mdata->reading[0] = buffer[i][0] * ACCEL_SCALE_M_S;
+            mdata->reading[1] = -buffer[i][1] * ACCEL_SCALE_M_S;
+            mdata->reading[2] = -buffer[i][2] * ACCEL_SCALE_M_S;
+        }
+#undef MAX_EPSILON
+#undef EPSILON_CHECK
+    }
+}
+
+static int
+accel_tick_do(struct accelerometer_adxl345_data *mdata)
+{
+    struct sol_direction_vector val =
+    {
+        .min = -ACCEL_RANGE,
+        .max = ACCEL_RANGE,
+        .x = mdata->reading[0],
+        .y = mdata->reading[1],
+        .z = mdata->reading[2]
+    };
+    int r;
+
+    accel_read(mdata);
+
+    r = sol_flow_send_direction_vector_packet
+            (mdata->node, SOL_FLOW_NODE_TYPE_ACCELEROMETER_ADXL345__OUT__OUT,
+            &val);
+
+    return r;
+}
+
+static void
+accel_ready(void *data)
+{
+    struct accelerometer_adxl345_data *mdata = data;
+
+    mdata->timer = NULL;
+    mdata->ready = true;
+
+    while (mdata->pending_ticks) {
+        accel_tick_do(mdata);
+        mdata->pending_ticks--;
+    }
+
+    SOL_DBG("accel is ready for reading");
+}
+
+static bool
+accel_init_stream(void *data)
+{
+    bool r;
+    struct accelerometer_adxl345_data *mdata = data;
+    static const uint8_t value = ACCEL_REG_FIFO_CTL_STREAM;
+
+    if (!sol_i2c_set_slave_address(mdata->i2c, ACCEL_ADDRESS)) {
+        SOL_WRN("Failed to set slave at address 0x%02x\n", ACCEL_ADDRESS);
+        return false;
+    }
+
+    /* enable FIFO in stream mode */
+    r = sol_i2c_write_register(mdata->i2c, ACCEL_REG_FIFO_CTL, &value, 1);
+    if (!r) {
+        SOL_WRN("could not set ADXL345 accel sensor's stream mode");
+        return false;
+    }
+
+    accel_ready(mdata);
+
+    return false;
+}
+
+static bool
+accel_init_rate(void *data)
+{
+    bool r;
+    struct accelerometer_adxl345_data *mdata = data;
+    static const uint8_t value = 0x0d;
+
+    if (!sol_i2c_set_slave_address(mdata->i2c, ACCEL_ADDRESS)) {
+        SOL_WRN("Failed to set slave at address 0x%02x\n", ACCEL_ADDRESS);
+        return false;
+    }
+
+    r = sol_i2c_write_register(mdata->i2c, ACCEL_REG_BW_RATE, &value, 1);
+    if (!r) {
+        SOL_WRN("could not set ADXL345 accel sensor's sampling rate");
+        return false;
+    }
+
+    if (accel_timer_resched(mdata, ACCEL_INIT_STEP_TIME,
```

Comment:
The first and last parameter of this function is always the same; since it's a `static` function, can't the last parameter be removed?